### PR TITLE
KAFKA-4885: Add client.close as exception handler in streams system tests

### PR DESCRIPTION
**Pull request by guozhangwang**

### CODE REVIEW

**Comment by guozhangwang** [comment]
system test passed: https://jenkins.confluent.io/job/kafka-streams-system-test-pr/16/console

**Review by dguy — APPROVED** [review]
LGTM

File: streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java [hunk 1]
```diff
@@ -634,6 +635,18 @@ public void close() {
         return new KafkaStreams(builder, props);
     }
 
+    private KafkaStreams createKafkaStreamsWithExceptionHandler(final KStreamBuilder builder, final Properties props) {
```
Comment (enothereska) [inline streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java]: Where is this used?

Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java]: my bad, just updated the PR.

**Comment by guozhangwang** [comment]
@mjsax @enothereska open for review again

File: streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java [hunk 2]
```diff
@@ -739,9 +740,21 @@ public void close() {
...
-        return new KafkaStreams(builder, props);
+        return createKafkaStreamsWithExceptionHandler(builder, props);
     }
 
+    private KafkaStreams createKafkaStreamsWithExceptionHandler(final KStreamBuilder builder, final Properties props) {
+        final KafkaStreams streamsClient = new KafkaStreams(builder, props);
+        streamsClient.setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+            @Override
+            public void uncaughtException(Thread t, Throwable e) {
+                streamsClient.close(30, TimeUnit.SECONDS);
```
Comment (hachikuji) [inline streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java]: Seems useful to know the exception. Will it get logged anywhere?

**Comment by guozhangwang** [comment]
@hachikuji

**Review by hachikuji — APPROVED** [review]
LGTM

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**